File: networking/src/main/java/ro/mpp2024/dto/DTOUtils.java
package ro.mpp2024.dto;

import ro.mpp2024.Employee;
import ro.mpp2024.Office;
import ro.mpp2024.Trip;
import ro.mpp2024.Destination;

public class DTOUtils {
    public static EmployeeDTO getEmployeeDTO(Employee employee) {
        return new EmployeeDTO(employee.getId(), employee.getUsername(), employee.getPassword(), employee.getOffice().getId());
    }

    public static Employee getEmployee(EmployeeDTO employeeDTO) {
        Office office = new Office(employeeDTO.getOfficeId(), null);
        return new Employee(employeeDTO.getId(), employeeDTO.getUsername(), employeeDTO.getPassword(), office);
    }

    public static TripDTO getTripDTO(Trip trip) {
        return new TripDTO(trip.getId(), trip.getDestination().getId(), trip.getDepartureDate(), trip.getDepartureTime(), trip.getAvailableSeats());
    }

    public static Trip getTrip(TripDTO tripDTO) {
        Destination destination = new Destination(tripDTO.getDestinationId(), null);
        return new Trip(tripDTO.getId(), destination, tripDTO.getDepartureDate(), tripDTO.getDepartureTime(), tripDTO.getAvailableSeats());
    }
}

File: csharpServer/networking/dto/DTOUtils.cs
using model;

namespace networking.dto;

public class DTOUtils
{
    public static EmployeeDTO GetEmployeeDTO(Employee employee)
    {
        return new EmployeeDTO(employee.Id, employee.Username, employee.Password, employee.Office.Id);
    }

    // public static Employee GetEmployee(EmployeeDTO employeeDTO)
    // {
    //     Office office = new Office(employeeDTO.OfficeId ?? 0, null);
    //     return new Employee(employeeDTO.Id, employeeDTO.Username, employeeDTO.Password, office);
    // }

    public static TripDTO GetTripDTO(Trip trip)
    {
        return new TripDTO(
            trip.Id,
            trip.Destination.Id,
            trip.DepartureDate,
            trip.DepartureTime,
            trip.AvailableSeats
        );
    }

    public static Trip GetTrip(TripDTO tripDTO)
    {
        Destination destination = new Destination(tripDTO.DestinationId, null);
        return new Trip(
            tripDTO.Id,
            destination,
            tripDTO.DepartureDate,
            tripDTO.DepartureTime,
            tripDTO.AvailableSeats
        );
    }
}